src/data/convert_to_mosiac.py

```diff
@@ -0,0 +1,69 @@
+from streaming import MDSWriter
+import huggingface_hub
+from datasets import load_dataset
+import argparse
+import os
+import tqdm
+
+
+def convert_to_mosiac_format(args):
+
+    # A dictionary mapping input fields to their data types
+    columns = {
+        'text': 'str',
+        'id': 'str'
+        # we can ignore the other fields, they're not used in training
+    }
+
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    with MDSWriter(out=args.output_dir, columns=columns, compression='zstd') as out:
+        # loop over our data
+        dataset = load_dataset(args.dataset, data_files={"train": "data/*"}, streaming=True)["train"]
+        for idx, instance in tqdm.tqdm(enumerate(dataset)):
+            out.write({
+                "text": instance["text"],
+                "id": instance["id"]
+            })
+
+
+    ### Example Usage
+    ## Upload
+    """
+    az storage blob upload-batch --source PATH_TO_SRC --destination https://ACCOUNT_NAME.blob.core.windows.net/bert24usablations --auth-mode login --destination-path data/NAME_OF_DATA
+    """
+    ## Download and Usage
+    """
+    # be sure to export AZURE_ACCOUNT_NAME and AZURE_ACCOUNT_ACCESS_KEY
```

Comment:
@NohTow This should work, although I don't know my AZURE_ACCOUNT_ACCESS_KEY to test this unfortunately. Will figure that out soon!

Reply:
@warner-benjamin the `AZURE_ACCOUNT_ACCESS_KEY` is the admin key for the "storage account" isn't it?

Reply:
Let's not be checking in specific container names. Use something like `https://<storageaccount>.blob.core.windows.net/<container>` and `azure://<container>/data/dolma_20b_stratified/` instead.

Reply:
Thanks @warner-benjamin, I wasn't sure which of these paths were things that should be kept secret. Updated the code and pushed to the EU data container instead of the US one (tagging @NohTow as an FYI).